src/main/java/org/springframework/integration/aws/inbound/kinesis/KclMessageDrivenChannelAdapter.java

```diff
@@ -364,26 +365,34 @@ public Duration waitPeriodToDeleteFormerStreams() {
 
 				};
 
-		private final List<StreamConfig> streamConfigs =
-				Arrays.stream(KclMessageDrivenChannelAdapter.this.streams)
-						.map(streamName ->
-								new StreamConfig(StreamIdentifier.singleStreamInstance(streamName),
-										KclMessageDrivenChannelAdapter.this.streamInitialSequence))
-						.toList();
+		private List<StreamConfig> streamConfigs = new ArrayList<>();
 
 		StreamsTracker() {
 		}
 
+		private StreamConfig buildStreamConfig(String streamName) {
+			logger.info(LogMessage.format("Initializing stream config for Stream: [%s] for Multi Stream Tracker", streamName));
+
+			DescribeStreamSummaryResponse streamSummaryResponse = KclMessageDrivenChannelAdapter.this.kinesisClient.describeStreamSummary(DescribeStreamSummaryRequest.builder().streamName(streamName).build()).join();
+			return new StreamConfig(StreamIdentifier.multiStreamInstance(Arn.fromString(streamSummaryResponse.streamDescriptionSummary().streamARN()), streamSummaryResponse.streamDescriptionSummary().streamCreationTimestamp().toEpochMilli()),
+					KclMessageDrivenChannelAdapter.this.streamInitialSequence);
+		}
+
 		@Override
 		public List<StreamConfig> streamConfigList() {
+			if (this.streamConfigs.size() == 0) {
```

Comment:
I think if `== null` is better and then you can use `Stream` API to produce a list after mapping to `StreamConfig`

Reply:
Yes, I agree, null check is better.